interop-testing/src/main/java/io/grpc/testing/integration/AbstractTransportTest.java

```diff
@@ -628,6 +634,48 @@ public void gracefulShutdown() throws Exception {
     verifyNoMoreInteractions(responseObserver);
   }
 
+  //-----------------------------------------------------------------------------
+  // The tests below are only work with gRpc interop test framework.
+  //-----------------------------------------------------------------------------
+
+  /**  Sends a large unary rpc with service account credentials. */
+  public void serviceAccountCreds(String jsonKey, InputStream credentialsStream, String authScope)
+      throws Exception {
+    ServiceAccountCredentials credentials =
+        (ServiceAccountCredentials) GoogleCredentials.fromStream(credentialsStream);
+    credentials = (ServiceAccountCredentials) credentials.createScoped(
+        Arrays.<String>asList(authScope));
+    TestServiceGrpc.TestServiceBlockingStub stub = TestServiceGrpc.newBlockingStub(channel)
+        .configureNewStub()
+        .addInterceptor(new ClientAuthInterceptor(credentials, testServiceExecutor))
+        .build();
+    final SimpleRequest request = SimpleRequest.newBuilder()
+        .setFillUsername(true)
+        .setFillOauthScope(true)
+        .setResponseSize(314159)
+        .setResponseType(PayloadType.COMPRESSABLE)
+        .setPayload(Payload.newBuilder()
+            .setBody(ByteString.copyFrom(new byte[271828])))
+        .build();
+
+    final SimpleResponse response = stub.unaryCall(request);
+    assertFalse(response.getUsername().isEmpty());
+    assertTrue("Received username: " + response.getUsername(),
+        jsonKey.indexOf(response.getUsername()) != -1);
+    assertFalse(response.getOauthScope().isEmpty());
+    assertTrue("Received oauth scope: " + response.getOauthScope(),
+        authScope.indexOf(response.getOauthScope()) != -1);
+
+    final SimpleResponse goldenResponse = SimpleResponse.newBuilder()
+        .setOauthScope(response.getOauthScope())
+        .setUsername(response.getUsername())
+        .setPayload(Payload.newBuilder()
+            .setType(PayloadType.COMPRESSABLE)
+            .setBody(ByteString.copyFrom(new byte[314159])))
+        .build();
+    assertEquals(goldenResponse, response);
```

Comment:
I'm concerned that this is too strict, since otherwise we wouldn't need to do the indexOf above.